src/QAT/qat/purr/integrations/qir.py

```diff
@@ -43,7 +43,7 @@ def __init__(
         self.result_variables = []
 
     def _get_qubit(self, id_: int):
-        return self.hardware.get_qubit(id_)
+        return self.hardware.qubits[id_]
```

Comment:
Index of qubit != index in list. What was wrong with the original way?

Reply:
The `_id` that gets passed to this method is a sequential integer, i.e. in a jagged hardware a qubit with that index may not exist.